platforms/native/language-native/src/main/java/org/gradle/swiftpm/plugins/SwiftPackageManagerExportPlugin.java

```diff
@@ -214,35 +215,36 @@ private void collectDependencies(Configuration configuration, Collection<Depende
                         throw new InvalidUserDataException(String.format("Cannot determine the Git URL for dependency on %s:%s.", dependency.getGroup(), dependency.getName()));
                     }
                     GitVersionControlSpec gitSpec = (GitVersionControlSpec) vcsSpec;
-                    dependencies.add(toSwiftPmDependency(externalDependency, gitSpec));
+                    gitSpec.getUrl().finalizeValue();
+                    dependencies.add(toSwiftPmDependency(externalDependency, gitSpec.getUrl().get()));
```

Comment:
💭  This place is being called lazily, so it is safe to resolve the property. We don't carry dependencies, but you cannot really have dependencies for your git url.